components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-list.jsp

```diff
@@ -30,166 +50,286 @@
-            url: 'idp-mgt-edit-finish-ajaxprocessor.jsp',
-            headers: {
-                Accept: "text/html"
-            },
-            data: 'idPName=' + encodeURIComponent(idpName) + '&enable=' + indicator,
-            async: false,
-            success: function (responseText, status) {
-                if (status == "success") {
-                    location.assign("idp-mgt-list-load.jsp");
+            try {
+                String cookie = (String) session.getAttribute(ServerConstants.ADMIN_SERVICE_COOKIE);
+                String backendServerURL = CarbonUIUtil.getServerURL(config.getServletContext(), session);
+                ConfigurationContext configContext =
+                            (ConfigurationContext) config.getServletContext().getAttribute(CarbonConstants.CONFIGURATION_CONTEXT);
+                IdentityProviderMgtServiceClient client = new IdentityProviderMgtServiceClient(cookie, backendServerURL, configContext);
+                int numberOfIdps = 0;
+                if (client.isPaginationSupported()) {
+                    if (hasFilter && !StringUtils.equals(filterString, DEFAULT_FILTER)) {
+                        numberOfIdps = client.getFilteredIdpCount(filterString);
+                        if (numberOfIdps > 0) {
+                            idpsToDisplay = client.getPaginatedIdpInfo(pageNumberInt + 1,
+                                        filterString);
+                        }
+                    } else {
+                        numberOfIdps = client.getIdpCount();
+
+                        if (numberOfIdps > 0) {
+                            if (numberOfIdps > resultsPerPageInt) {
+                                idpsToDisplay = client.getAllPaginatedIdpInfo(pageNumberInt + 1);
+
+                            } else {
+                                idpsToDisplay = client.getIdPs();
+                            }
+                        }
+                    }
+                } else {
+                    IdentityProvider[] idps = client.getIdPs();
+                    if (idps != null) {
+                        numberOfIdps = idps.length;
+                        int startIndex = pageNumberInt * resultsPerPageInt;
+                        int endIndex = (pageNumberInt + 1) * resultsPerPageInt;
+                        idpsToDisplay = new IdentityProvider[resultsPerPageInt];
+
+                        for (int i = startIndex, j = 0; i <= endIndex && i < idps.length; i++, j++) {
+                            idpsToDisplay[j] = idps[i];
+                        }
+                    }
                 }
+                numberOfPages = (int) Math.ceil((double) numberOfIdps / resultsPerPageInt);
+                Map<String, UUID> idpUniqueIdMap = new HashMap<String, UUID>();
+
+                if(idpsToDisplay != null) {
+                    for (IdentityProvider provider : idpsToDisplay) {
+                       idpUniqueIdMap.put(provider.getIdentityProviderName(), UUID.randomUUID());
```

Comment:
what is the requirement of the idpUniqueIdMap?

Reply:
We use idpUniqueIdMap for idpEdit and many other places also
https://github.com/wso2/carbon-identity-framework/blob/13d125060e769c08efdaf2c2945bdc425c4e3e57/components/idp-mgt/org.wso2.carbon.idp.mgt.ui/src/main/resources/web/idpmgt/idp-mgt-edit.jsp#L215